crates/ruff_linter/src/rules/flake8_use_pathlib/rules/dotless_with_suffix.rs

```diff
@@ -0,0 +1,104 @@
+use crate::checkers::ast::Checker;
+use ruff_diagnostics::{AlwaysFixableViolation, Diagnostic, Edit, Fix};
+use ruff_macros::{derive_message_formats, ViolationMetadata};
+use ruff_python_ast::{Expr, ExprAttribute, ExprCall, ExprStringLiteral, StringFlags};
+use ruff_python_semantic::analyze::typing;
+use ruff_python_semantic::SemanticModel;
+use ruff_text_size::Ranged;
+
+/// ## What it does
+/// Checks for `Path.with_suffix()` calls where
+/// the given suffix does not have a leading dot.
+///
+/// ## Why is this bad?
+/// `Path.with_suffix()` will raise an error at runtime
+/// if the given suffix is not prefixed with a dot.
+///
+/// ## Known problems
+/// This rule is prone to false positives and negatives
+/// due to type inference limitations.
```

Comment:
We should be more explicit about what the possible errors are here. As a reader, it's not clear to me what that means (what's even type inference)? What are the cases I have to worry about.